src/realm/mixed.hpp

```diff
@@ -254,14 +254,24 @@ inline Mixed::Mixed(bool v) noexcept
 
 inline Mixed::Mixed(float v) noexcept
 {
-    m_type = int(type_Float) + 1;
-    float_val = v;
+    if (null::is_null_float(v)) {
+        m_type = 0;
+    }
+    else {
+        m_type = int(type_Float) + 1;
+        float_val = v;
+    }
 }
 
 inline Mixed::Mixed(double v) noexcept
 {
-    m_type = int(type_Double) + 1;
-    double_val = v;
+    if (null::is_null_float(v)) {
```

Comment:
how about for the `Optional<double>` and `Optional<float>` constructors below?

Reply:
Fixed